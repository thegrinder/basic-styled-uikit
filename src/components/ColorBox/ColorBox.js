import React from 'react';
import { string, number } from 'prop-types';

import StyledColorBox from './StyledColorBox';

const propTypes = {
  /** one of: neutral, primary, success, danger, warning */
  bgColor: string,
  /** one of: 1, 2 */
  elevation: number,
};

const ColorBox = props => <StyledColorBox {...props} />;

ColorBox.propTypes = propTypes;

export default ColorBox;
