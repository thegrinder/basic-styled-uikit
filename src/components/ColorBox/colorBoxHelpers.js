import { css } from 'styled-components';

export const getBoxBgColor = (theme, bgColor) => (
  theme.uiKit.colorBox.bgColors[bgColor]
);
export const colorBoxBgStyle = ({ theme, bgColor }) => (
  css`background-color: ${getBoxBgColor(theme, bgColor)};`
);

export const getBoxBorderColor = (theme, borderColor) => (
  theme.uiKit.colorBox.borderColors[borderColor]
);
export const colorBoxBorderStyle = ({ theme, borderColor }) => (
  css`border-color: ${getBoxBorderColor(theme, borderColor)};`
);
