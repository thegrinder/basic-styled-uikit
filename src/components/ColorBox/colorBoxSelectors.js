import { getColorBox } from '../../theme/themeSelectors';

export const getBoxBgColor = (theme, bgColor) => (
  getColorBox(theme).bgColors[bgColor]
);

export const getBoxElevation = (theme, elevation) => (
  getColorBox(theme).elevations[elevation]
);
