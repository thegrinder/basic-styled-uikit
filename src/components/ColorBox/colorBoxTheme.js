const colorBoxTheme = {
  bgColors: {
    transparent: 'transparent',
    white: '#fff',
    lightGray: '#f8f8f8',
    gray: '#666',
    lightPrimary: '#d8eafc',
    primary: '#1e87f0',
    lightSuccess: '#edfbf6',
    success: '#32d296',
    lightWarning: '#fff6ee',
    warning: '#faa05a',
    lightDanger: '#fef4f6',
    danger: '#f0506e',
  },
  borderColors: {
    transparent: 'transparent',
    white: '#fff',
    gray: '#e5e5e5',
    primary: '#1e87f0',
    success: '#32d296',
    warning: '#faa05a',
    danger: '#f0506e',
  },
};

export default colorBoxTheme;
