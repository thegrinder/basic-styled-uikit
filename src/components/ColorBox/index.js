export { default as ColorBoxProvider } from './ColorBoxProvider';
export { default as ColorBox } from './ColorBox';
export { useColorBoxContext } from './context';
