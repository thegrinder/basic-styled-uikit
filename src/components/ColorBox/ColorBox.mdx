---
name: ColorBox
menu: Components
---

import { Playground, Props } from 'docz';
import ColorBox from './ColorBox';
import Text from '../Text/Text';
import { Spacer } from '../../../docz';

# ColorBox

<Props of={ColorBox} />


### ColorBox backgrounds

<Playground>
  <ColorBox bgColor="primary">
    <Text color="neutral1000">primary</Text>
  </ColorBox>
  <Spacer />
  <ColorBox bgColor="danger">
    <Text color="neutral1000">danger</Text>
  </ColorBox>
  <Spacer />
  <ColorBox bgColor="success">
    <Text color="neutral1000">success</Text>
  </ColorBox>
  <Spacer />
  <ColorBox bgColor="warning">
    <Text color="neutral1000">warning</Text>
  </ColorBox>
  <Spacer />
  <ColorBox bgColor="neutral0">
    <Text color="neutral1000">neutral0</Text>
  </ColorBox>
</Playground>


### ColorBox elevation

<Playground>
  <ColorBox elevation={1}>one</ColorBox>
  <Spacer />
  <ColorBox elevation={2}>two</ColorBox>
</Playground>
