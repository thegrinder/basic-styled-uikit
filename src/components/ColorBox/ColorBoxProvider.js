import { createProvider } from '../../helpers';
import { colorBoxBaseTheme, colorBoxDarkTheme } from './colorBoxTheme';

const ColorBoxProvider = createProvider(
  'colorBox',
  colorBoxBaseTheme,
  colorBoxDarkTheme
);

export default ColorBoxProvider;
