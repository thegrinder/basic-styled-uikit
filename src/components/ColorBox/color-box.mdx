---
name: ColorBox
menu: Components
route: /components/color-box
---

import { Playground, Props } from 'docz';
import ColorBoxProvider from './ColorBoxProvider';
import ColorBox from './ColorBox';
import { Text } from '../Typography';
import { Spacer } from '../../../docz';

## **ColorBoxProvider**

`<ColorBoxProvider />` is responsible for passing down the theme to the `<ColorBox />` component. It expects the theme to be of certain structure. You can use any valid css properties written in camel case.


### **Example**

The minimal color box theme has to include one background color and one elevation variation.

<Playground>
  {() => {
    const colorBoxTheme = {
      bgColors: {
        primary: '#1e88e5',
      },
      elevations: {
        1: {
          boxShadow:
            '0 3px 1px -2px rgba(0,0,0,.2), 0 2px 2px 0 rgba(0,0,0,.14), 0 1px 5px 0 rgba(0,0,0,.12)',
        },
      },
    };

    const Example = () => (
      <ColorBoxProvider theme={colorBoxTheme}>
        <>
          <ColorBox bgColor="primary">background color primary</ColorBox>
          <Spacer />
          <ColorBox elevation={1}>elevation one</ColorBox>
        </>
      </ColorBoxProvider>
    );

    return <Example />;
  }}
</Playground>


However, you can easily add as many background colors and variations as you wish.

<Playground>
  {() => {
    const colorBoxTheme = {
      bgColors: {
        neutral: '#fff',
        primary: '#1E88E5',
        success: '#43A047',
        warning: '#FDD835',
        danger: '#F4511E',
      },
      elevations: {
        1: {
          boxShadow:
            '0 3px 1px -2px rgba(0,0,0,.2), 0 2px 2px 0 rgba(0,0,0,.14), 0 1px 5px 0 rgba(0,0,0,.12)',
        },
        2: {
          boxShadow:
            '0 2px 4px -1px rgba(0,0,0,.2), 0 4px 5px 0 rgba(0,0,0,.14), 0 1px 10px 0 rgba(0,0,0,.12)',
        },
      },
    };

    const Example = () => (
      <ColorBoxProvider theme={colorBoxTheme}>
        <>

          <ColorBox bgColor="primary">
            primary
          </ColorBox>
          <Spacer />
          <ColorBox bgColor="danger">
            danger
          </ColorBox>
          <Spacer />
          <ColorBox bgColor="success">
            success
          </ColorBox>
          <Spacer />
          <ColorBox bgColor="warning">
            warning
          </ColorBox>
          <Spacer />
          <ColorBox bgColor="neutral">
            neutral
          </ColorBox>
          <Spacer />

          <ColorBox bgColor="neutral" style={{ padding: '20px' }}>
            <ColorBox elevation={1} style={{ padding: '20px' }}>
              <span>one</span>
              <ColorBox elevation={1} style={{ padding: '20px' }}>
                <span>one</span>
                <ColorBox elevation={1} style={{ padding: '20px' }}>
                  <span>one</span>
                </ColorBox>
              </ColorBox>
            </ColorBox>
            <Spacer />
            <ColorBox elevation={2}>
              <span>two</span>
            </ColorBox>
          </ColorBox>

        </>
      </ColorBoxProvider>
    );

    return <Example />;
  }}
</Playground>

---

## **ColorBox**

ColorBox component grabs the theme given to [ColorBoxProvider](/basic-styled-uikit/components/color-box#colorboxprovider) and applies the style defined there based on the given `bgColor` and `elevation` prop value.

<Props of={ColorBox} />
