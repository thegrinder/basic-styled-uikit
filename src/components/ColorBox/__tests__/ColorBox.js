import React from 'react';
import { render } from '../../../test-utils';
import ColorBox from '../ColorBox';

const children = 'children';

const renderComponent = (props = {}) =>
  render(<ColorBox {...props}>{children}</ColorBox>);

describe('<ColorBox />', () => {
  it('should render correctly with default props and children', () => {
    const { queryByText } = renderComponent();
    expect(queryByText(children)).toBeTruthy();
  });

  it('should render correctly with custom props', () => {
    const { queryByText } = renderComponent({
      bgColor: 'primary',
      borderColor: 'success',
    });
    expect(queryByText(children)).toBeTruthy();
  });

  it('should render <div> tag', () => {
    const { queryByText } = renderComponent();
    expect(queryByText(children).tagName).toEqual('DIV');
  });
});
