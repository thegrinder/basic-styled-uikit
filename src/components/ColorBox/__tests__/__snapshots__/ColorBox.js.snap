// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorBox /> should render <div> tag 1`] = `
<div
  class="StyledColorBox-sc-1ihbgka-0 hmqRpq"
>
  children
</div>
`;

exports[`<ColorBox /> should render correctly with custom props 1`] = `
<div
  class="StyledColorBox-sc-1ihbgka-0 jxpoGf"
>
  children
</div>
`;

exports[`<ColorBox /> should render correctly with default props and children 1`] = `
<div
  class="StyledColorBox-sc-1ihbgka-0 hmqRpq"
>
  children
</div>
`;
